### ISSUE TITLE
Replace string id feature

### ISSUE THREAD

**Issue by ismaelgomescosta** [javers/javers#614]
I need a feature to override tha method called to create the id of the entity.

I have a composite id with entities embbeded. So I created a new annotation to customize Javers to use a defined method not toString Object.

So before Javers created globalId like = 'dummyEntityWithCompositeEmbeddedId/org.javers.core.model.DummyPoint@5b10,1'

and with this new feature I can customize for = 'dummyEntityWithCompositeEmbeddedId/1,2,1'

Here is my pull request #613. Sorry about my last issues. I should read the CONTRIBUTING.md before

**Comment by bartoszwalacik** [comment]
Did you tried to implement custom Type Adapter ?
see https://javers.org/documentation/domain-configuration/#ValueType

**Comment by mundodojava** [comment]
My difficulty isn't the serialization of cdoId. My problem is the serialization of globalId_key field. 
And this is part of javers core. I know my feature can be looks like a personal necessity but if you run my tests without the new feature you will see globalId_key is equal  'dummyEntityWithCompositeEmbeddedId/org.javers.core.model.DummyPoint@5b10,1'

My worry is about maintenance of snapshots. I will never can find by globalId_key because it was saved with object reference not with my real values.

**Comment by bartoszwalacik** [comment]
ok , got it, that's resonable

**Comment by mundodojava** [comment]
Thank you. Please if it's all right can you merge this feature? I need to use that next week in my job. 

Thanks for your patience

**Comment by bartoszwalacik** [comment]
will try , but CR first

**Comment by bartoszwalacik** [comment]
I have looked at the PR, why another annotation? If you can add annotation to this class why you cant just fix its `toString()`?

Btw, sometimes you can't modify a class (think about legacy code or libraries code).

I think about more versatile and simpler solution with Function:

API (JaversBuilder)

```
public <T> void registerToStringFunction(Class<T> clazz, Function<T, String> toString);
```

usage
```
javersBuilder.registerToStringFunction(YourClass.class, YourClass::customTtoString);
```

**Comment by mundodojava** [comment]
I agree with you, it's a better solution then my. I will adjust the code

**Comment by bartoszwalacik** [comment]
fine :)

**Comment by mundodojava** [comment]
I made the changes you suggested. Can you review my code again please?

**Comment by mundodojava** [comment]
I'm still work in a solution, as soon as possible I will commit it. Tks :)

**Comment by bartoszwalacik** [comment]
ok, let me know when you will be ready :)

**Comment by mundodojava** [comment]
Hi, I finished my changes but I don`t know how to make travis run again with my branch. Because I already fix the conflicts. Can you review for me please?

**Comment by bartoszwalacik** [comment]
Guys, I appreciate your effort. I did some changes to the API and impl to make the code more *javers-way*. Take a look at the tests, is it OK now?

**Comment by bartoszwalacik** [comment]
Ismael Gomes Costa , your commits are not bounded to your github profile, if you bound them, you will be mentioned as a JaVers contributor. You can do it by adding the email attached to your commits to your github profile

**Comment by mundodojava** [comment]
Ok, I will do it. Thanks

**Comment by mundodojava** [comment]
I agree with your changes. But I didn't understand why the PR is not passing. Is this because any of the changes or is just because something wrong with travis?

**Comment by bartoszwalacik** [comment]
that's strange, looks like sth is wrong with  github-travis integrations, travis builds seems fine https://travis-ci.org/javers/javers/builds

**Comment by bartoszwalacik** [comment]
@ismaelgomescosta forks are good for occasional contrubitors. If you'd like to contribute more, I can give you access to a branch in javers repository

**Comment by ismaelgomescosta** [comment]
Ok. If I can I will try to contribute more. I like this project. I always had problems with envers and all these rows in the database. Now I use Javers with mongo and I'm happy with this.

**Comment by bartoszwalacik** [comment]
@ismaelgomescosta great! So what's next? (btw. 3.7.6 is on its way to Central)

**Comment by ismaelgomescosta** [comment]
Thanks. At my job we are thinking change mongodb for dynamodb so I would like to contribute with a dynamodb module. I  saw you have an issue open for this

**Comment by bartoszwalacik** [comment]
good idea. if you want we can work together on data model design

**Comment by bartoszwalacik** [comment]
released in 3.7.6

**Comment by hernaniturpin** [comment]
I could not find a way to register a class providing a function to define the id. I need this becaus I am using classes that I can not change.
The solution suggested by @bartoszwalacik with the agreement of @ismaelgomescosta will solve my issue. According to this issues this was implemented in 3.7.6. But I can not find it in 5.1.2. Am I misssing something?

**Comment by bartoszwalacik** [comment]
usage example
https://javers.org/javadoc_5.x/org/javers/core/JaversBuilder.html#registerValueWithCustomToString-java.lang.Class-java.util.function.Function-
